src/gui/symbology/qgssymbolslistwidget.cpp

```diff
@@ -299,6 +304,47 @@ void QgsSymbolsListWidget::showAnimationSettings()
   }
 }
 
+void QgsSymbolsListWidget::showExtentBufferSettings()
+{
+  QgsPanelWidget *panel = QgsPanelWidget::findParentPanel( this );
+  if ( panel && panel->dockMode() )
+  {
+    QgsExtentBufferWidget *widget = new QgsExtentBufferWidget( mSymbol, mLayer, panel );
+    widget->setPanelTitle( tr( "Extent Buffer" ) );
+    widget->setContext( mContext );
+
+    connect( widget, &QgsPanelWidget::widgetChanged, this, [ = ]()
+    {
+      mSymbol->setExtentBuffer( widget->extentBuffer() );
+      mSymbol->setDataDefinedProperty( QgsSymbol::Property::ExtentBuffer, widget->dataDefinedProperty() );
+      mSymbol->setExtentBufferSizeUnit( widget->sizeUnit() );
+
+      emit changed();
+    } );
+
+    panel->openPanel( widget );
+
+  }
+  else
+  {
+    QgsExtentBufferDialog dlg( mSymbol, mLayer, panel );
```

Comment:
We should also have a "setContext" call here for the QgsExtentBufferDialog class, which in turn sets it for the QgsExtentBufferWidget. That'll give us consistency if this dialog is opened via the layer properties window.